skimage/exposure/hdr.py

```diff
@@ -0,0 +1,311 @@
+import numpy as np
+
+
+def get_radiance(images, exposure, radiance_map):
+    """
+    Return the radiance image for a series of images based upon a camera
+    response function.
+
+    Parameters
+    ----------
+    images: list
+        List of images in the for of numpy arrays. Either mono or color
+        (RGB, MxNx3).
+    exposure : numpy 1D array
+        Array of exposure times in seconds.
+    radiance_map : numpy array
+        Array mapping the counts to radiance
+
+    Returns
+    -------
+    hdr : numpy array
+        Resulting image with radiance values.
+    """
+
+    den = np.ones(images[0].shape)
+    num = np.zeros(images[0].shape)
+    wf = np.vectorize(_weight_func)
+    for idx, im in enumerate(images):
+        gij = im.copy()
+        # For colour
+        if im.ndim == 3:
+            for ii in range(im.shape[2]):
+                gij[:, :, ii] = radiance_map[im[:, :, ii] + 1, ii]
+        else:
+            gij[:, :] = radiance_map[im[:, :, ii] + 1]
+        gij = gij - np.log(exposure[idx])
+
+        wij = wf(gij)
+        num = num + wij * gij
+        den = den + wij
+
+    return num / den
+
+
+def make_hdr(images, exposure, radiance_map, depth=16):
+    """
+    Compute the HDR image from a series of images with a given radiance
+    mapping.
+
+    Parameters
+    ----------
+    images: list or ImageCollection
+        List of images as numpy arrays. Either grayscale or color
+        (RGB).
+        Can also be an ImageCollection
+    exposure : numpy 1D array
+        Array of exposure times in seconds.
+        Images some times have these in the exif information.
+    radiance_map : numpy array
+        Array (idx) mapping counts to radiance value, if input is RGB this must
+        be Nx3. See get_crf
+    depth : int, optional
+        Pixel depth.
+
+    Returns
+    -------
+    hdr : numpy array
+        The HDR image, either grayscale or RGB depending on input, in ln(E).
+
+    References
+    ----------
+    .. [1]  Debevec, P. E., & Malik, J. (1997). SIGGRAPH 97 Conf. Proc.,
+            August, 3-8. DOI:10.1145/258734.258884
+    """
+    B = np.log(np.array(exposure))
+
+    if images[0].ndim == 3:
+        sx, sy, sc = np.shape(images[0])
+        hdr = np.zeros([sx, sy, sc], dtype=np.float)
+        gray = False
+    else:
+        sx, sy = np.shape(images[0])
+        hdr = np.zeros([sx, sy], dtype=np.float)
+        gray = True
+
+    images = np.asarray(images, dtype=np.uint64)
+
+    sx, sy, sz = np.shape(images[0])
+    w = _weight_func_arr(images, depth)
+
+    if gray:
+        num = np.zeros_like(hdr)
+        den = np.zeros_like(hdr)
+
+        for kk in range(images.shape[0]):
```

Comment:
Is this doing a line-by-line scan?

Reply:
No, it runs through the list of differently exposed images to produce the hdr image (equation 6 in the referenced paper).
I see that there could be more comments, and I can try and add some if this goes forward